File: src/problems/leetcode-cn/1. 两数之和/solution/python/solution.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        m = {}
        for i, v in enumerate(nums):
            t = target - v
            if t in m:
                return [m[t], i]
            else:
                m[v] = i
        return [0, 0]

File: src/problems/leetcode-cn/1. 两数之和/solution/go/solution.go
func twoSum(nums []int, target int) []int {
    length, m := len(nums), make(map[int]int)
    for i := 0; i < length; i++ {
        t := target - nums[i]
        if v, ok := m[t]; ok {
            return []int{v, i}
        } else {
            m[nums[i]] = i
        }
    }
    return []int{0, 0}
}